bin/proving-service/.env

```diff
@@ -11,4 +11,5 @@ MPS_HEALTH_CHECK_INTERVAL_SECS="1"
 MPS_PROMETHEUS_HOST="127.0.0.1"
 MPS_PROMETHEUS_PORT="6192"
 MPS_ENABLE_METRICS="false"
+MPS_PROXY_UPDATE_URL="http://0.0.0.0:8083"
```

Comment:
I was thinking this would affect all similar variables - e.g.:

- `MPS_URL` instead of `MPS_HOST` and `MPS_PORT`.
- `MPS_PROMETHEUS_URL` instead of `MPS_PROMETHEUS_HOST` and `MPS_PROMETHEUS_PORT`.

Also, it is not clear to me why we need both `MPS_PROXY_UPDATE_URL` and `MPS_WORKERS_UPDATE_PORT`. Assuming we have a single `MPS_URL`, I think it probably makes sense to keep `MPS_WORKERS_UPDATE_PORT` and not introduce `MPS_PROXY_UPDATE_URL`.

Reply:
> Also, it is not clear to me why we need both `MPS_PROXY_UPDATE_URL` and `MPS_WORKERS_UPDATE_PORT`. Assuming we have a single `MPS_URL`, I think it probably makes sense to keep `MPS_WORKERS_UPDATE_PORT` and not introduce `MPS_PROXY_UPDATE_URL`.

Thought, in this case, I'm not sure how it'll work with different command variants.

Reply:
I don't have a strong preference here. @igamigo @Mirko-von-Leipzig - what do you guys think?

Reply:
imo it depends on the desired usage.

For something the app is hosting itself it really only has `localhost` or `0.0.0.0` plus a port. In the case of a proxy I imagine these would _always_ be `0.0.0.0`? In which case only specifying ports it enough. And if ports are enough then this thread is sort of moot :)

I guess another question is should their be separate ports at all? Is this not the management interface - and that should just be a single port/host? But maybe with different paths?

If they must remain separate then its much of a muchness to me :)

Reply:
Yeah, doesn't seem like URL would ever change from `0.0.0.0`. In that sense I agree ports would be enough. In terms of the original question I think I'd favor leaving a single host var and different ports which is similar to what was already there, all other options seem to be more ambiguous.

> I guess another question is should their be separate ports at all? Is this not the management interface - and that should just be a single port/host? But maybe with different paths?

I think I'd definitely prefer serving the management plane over a different port than the one meant for user traffic.

Reply:
> I think I'd definitely prefer serving the management plane over a different port than the one meant for user traffic.

Ah I misunderstood. Yes definitely separate ports 😁

Reply:
So, would the conclusion be something like?:

- We get rid of `MPS_HOST` but keep both `MPS_PORT` and `MPS_WORKER_UPDATE_PORT` (maybe rename this to `MPS_CONTROL_PORT`?).
  - This would also imply getting rid of the newly introduced `MPS_PROXY_UPDATE_URL`.
- We combine `MPS_PROMETHEUS_HOST` and `MPS_PROMETHEUS_PORT` into `MPS_PROMETHEUS_URL`.

Reply:
Awesome, thanks for the replies. I'm using this new approach.

Reply:
Since we use the same server to serve the metrics, I consider that removing the `MPS_PROMETHEUS_HOST` was the way to go. Instead of creating `MPS_PROMETHEUS_URL`, I leave `MPS_PROMETHEUS_PORT` and used the same host as the other urls.

Reply:
Would we never want to have Prometheus running a different server?

Reply:
What we have in the same server is the exporter, that collects metrics from the proxy and formats it prometheus-compatible and serves it in the some port using the `/metrics` endpoint. Then any Prometheus Collector can scrape it to store it in a time-series DB and then be provided to a dashboard.

Reply:
So, basically, we have the proxy running a single process but we can communicate with it via 3 ports:

- The proxy port: this is used for regular proving requests.
- The control port: this is used for adding/removing workers.
- The prometheus port: this could be used by a prometheus collector to periodically get metrics from the proxy.

Is this correct?

Reply:
That's correct.